src/stirling/bpf_tools/probe_specs/probe_specs.h

```diff
@@ -75,6 +75,9 @@ struct KProbeSpec {
   // kernels.
   bool is_optional = false;
 
+  // If the kernel function is not found, then this fallback function will be used instead.
+  struct KProbeSpec* fallback_probe = nullptr;
```

Comment:
This should be a managed ptr (`std::unique_ptr`) otherwise the memory will be leaked when the struct is deallocated.